### PR TITLE
tsdb-series enhancements

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR adds Json output to `tsdb-series` and also implements filtering of samples when computing per-series data-points per minute.

#### Checklist

- [na] Tests updated.
- [na] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [na] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: tools/tsdb-series/main.go [hunk 1]
```diff
@@ -44,6 +51,13 @@ func main() {
 	}
 	ctx := context.Background()
 
+	if !time.Time(minTime).IsZero() {
+		level.Info(logger).Log("msg", "using minTime for samples", "minTime", formatTime(time.Time(minTime)))
+	}
+	if !time.Time(maxTime).IsZero() {
+		level.Info(logger).Log("msg", "using maxTime for samples", "minTime", formatTime(time.Time(maxTime)))
```
Comment (dimitarvdimitrov) [inline tools/tsdb-series/main.go:58]:
```suggestion
		level.Info(logger).Log("msg", "using maxTime for samples", "maxTime", formatTime(time.Time(maxTime)))
```

File: tools/tsdb-series/main.go [hunk 2]
```diff
@@ -130,19 +168,50 @@ func printBlockIndex(ctx context.Context, blockDir string, printChunks bool, ser
...
-					"max time:", c.MaxTime, formatTime(timestamp.Time(c.MaxTime)))
+				output.Chunks = append(output.Chunks, chunk{
+					Ref:     c.Ref,
+					MinTime: c.MinTime,
+					MaxTime: c.MaxTime,
+				})
+			}
+		}
+
+		if jsonOutput {
```
Comment (dimitarvdimitrov) [inline tools/tsdb-series/main.go:192]: _this a nitpick on a maybe throwaway code, so feel free to ignore_, but i'd maybe pull out the printing logic into something like a `printer` which can be implemented by an `stdoutPrinter` and a `jsonPrinter` to keep `printBlockIndex` shorter.

**Review by dimitarvdimitrov — APPROVED** [review]

File: tools/tsdb-series/main.go [hunk 3]
```diff
@@ -157,38 +226,53 @@ func formatTime(ts time.Time) string {
...
+				include = false
+			}
+			if maxTimeFilter > 0 && ts > maxTimeFilter {
+				include = false
+			}
+			if include {
+				totalSamples++
+				if minTime == 0 {
+					minTime = ts
+				}
```
Comment (seizethedave) [inline tools/tsdb-series/main.go:265]: Is it a given that we're iterating in increasing time order?

**Review by seizethedave — APPROVED** [review]
LGTM%C 😃

File: tools/tsdb-series/main.go (hunk 3 above)
Comment (pstibrany) [reply tools/tsdb-series/main.go:265]: TSDB library produces chunks with samples with increasing timestamps, so it *should* be.

**Pull request merged**